app/controllers/departments_controller.rb

```diff
@@ -50,24 +51,7 @@ def set_department
     @department = Department.find(params[:id])
   end
 
-  def phone_attrs
-    [:id, :body, :label, :_destroy, :type, :contacts_id]
-  end
-
-  def email_attrs
-    [:id, :body, :label, :_destroy, :type, :contacts_id]
-  end
-
-  def contact_attrs
-    [
-      :id, :name, :_destroy, :contactable_id, :contactable_type, :street,
-      :extended, :city, :postal_code,
-      contact_emails_attributes: email_attrs,
-      contact_phones_attributes: phone_attrs
-    ]
-  end
-
   def department_params
-    params.require(:department).permit(user_ids: [], contact_attributes: contact_attrs)
+    params.require(:department).permit(policy(@department).permitted_attributes)
```

Comment:
and not evaluating such things is just very bad practice.
If I don't evaluate this, a department_manager can just add a select manually to his department form within his browsers inspect:
```html
<select name="department[user_ids][]" id="department_user_ids">
  <option value="666"></option>
</select>
```
Submission will just be accepted and commited to the database. Maybe rails will evaluate that the id 666 is not existing. But if I hit an existing id, it will be in the database.
This is just the simplest method, there would be much tougher stuff that could be done. This could be accomplished by anyone that knows what a Browsers inspector is and knows how to insert a select into his form there.

You can well argue that this is an edge case, and you want to trust the users and so on. 
But this is just **extremely bad practice** and sooner or later it will flash back with a breach that will be at least very embarrassing, if not harmful.